Add jdbc connector e2e test

zhangyuge1 opened the pull request:
## Purpose of this pull request

related issue: #2171

## Check list

* [ ] Code changed are covered with tests, or it does not need tests for reason:
* [ ] If any new Jar binary package adding in your PR, please add License Notice according
  [New License Guide](https://github.com/apache/incubator-seatunnel/blob/dev/docs/en/contribution/new-license.md)
* [ ] If necessary, please update the documentation to describe the new feature. https://github.com/apache/incubator-seatunnel/tree/dev/docs

zhangyuge1 committed:
add jdbc e2e test

zhangyuge1 committed:
add jdbc e2e test

CalvinKirs commented on seatunnel-core/seatunnel-flink-starter/pom.xml:
```diff
@@ -103,6 +103,17 @@
             <artifactId>junit</artifactId>
         </dependency>
 
+        <dependency>
+            <groupId>mysql</groupId>
+            <artifactId>mysql-connector-java</artifactId>
+            <scope>compile</scope>
+        </dependency>
```
we can't use MySQL in the complier step.

CalvinKirs replied:
@ruanwenjun could you give @zhangyuge1 some suggestions?

zhangyuge1 replied:
But if I don't use mysql in the compiler step,it can not find jdbc driver.

ruanwenjun replied:
Yes, we cannot use mysql, we can use other db to test the jdbc case e.g. clickhouse.

dinggege1024 replied:
> 

Hi would show me the e.g. clickhouse project code path?

dinggege1024 replied:
> Yes, we cannot use mysql, we can use other db to test the jdbc case e.g. clickhouse.

Hi would you show me the e.g. clickhouse code path?

CalvinKirs commented on seatunnel-e2e/seatunnel-flink-connector-v2-e2e/src/test/java/org/apache/seatunnel/e2e/flink/v2/jdbc/FakeSourceToJdbcIT.java:
```diff
@@ -0,0 +1,79 @@
+import org.junit.After;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
```
you need add License headers.

zhangyuge1 committed:
add jdbc(postgresql) e2e test

zhangyuge1 committed:
add jdbc(postgresql) e2e test

zhangyuge1 committed:
Merge branch 'apache:dev' into dev

zhangyuge1 committed:
add pdjdbc license

zhangyuge1 committed:
add pdjdbc license

zhangyuge1 committed:
Merge branch 'apache:dev' into dev

zhangyuge1 committed:
Merge branch 'dev' of https://github.com/zhangyuge1/incubator-seatunnel into dev

zhangyuge1 committed:
add pgjdbc license

zhangyuge1 committed:
add pgjdbc license

zhangyuge1 committed:
fix vcs.xml

zhangyuge1 committed:
fix vcs.xml

TyrantLucifer committed:
[Connector-V2] Add parquet file reader for Hive Source Connector (#2199) (#2237)

* [Connector-V2][connector-hive] Add parquet read strategy in hive connector source

* [Connector-V2][connector-hive] Optimize the process of generating file read strategies

zhaomin1423 committed:
fix wrong comment (#2249)

ashulin committed:
[api] improve config options (#2179)

* [api] improve config options

* [api] option test & new config

* [api] option check style

* [api] support complex type option

* [api] deprecated synchronized & simplified equlas

A read-only configuration don't require the 'synchronized' keyword.
hashmap initialcapacity: https://cloud.tencent.com/developer/article/1913508

* [api] suppress magic number warning

zhuangchong committed:
[Connector-V2][Doc]add new socket source doc (#2205)

* add new socket source doc

* add example in the socket source doc

zhuangchong committed:
.gitignore file adds log keyword to ignore. (#2271)

zhangyuge1 committed:
add checker-qual-3.5.0 license

zhangyuge1 committed:
add checker-qual-3.5.0 license

TyrantLucifer committed:
[Connector-V2] Add parquet writer in file connector (#2273)

songjianet committed:
[Feat][UI] Add the modal in the user manage. (#2277)

dijiekstra committed:
[seatunnel-1947][seatunnel-server] add scheduler spi & integration with dolphinscheduler. (#2190)

CalvinKirs commented on seatunnel-core/seatunnel-flink-starter/pom.xml:
```diff
@@ -103,6 +103,17 @@
+            <version>${project.version}</version>
+        </dependency>
+
+        <dependency>
+            <groupId>org.postgresql</groupId>
```
If necessary, I recommend putting it in the JDBC dependency, and PG will be the default implementation we provide.

zhangyuge1 replied:
If put it in the pom of jdbc connector, it doesn't work because the driver could not be found.

CalvinKirs replied:
did I miss something?  could you try it?

zhangyuge1 replied:
I have tried it.

CalvinKirs replied:
https://github.com/apache/incubator-seatunnel/blob/dev/seatunnel-connectors-v2/connector-jdbc/pom.xml#L49 We will use it as the default driver, so we need to change its scope

CalvinKirs commented on tools/dependencies/known-dependencies.txt:
```diff
@@ -732,4 +732,6 @@ zookeeper-jute-3.5.9.jar
\ No newline at end of file
+jakarta.xml.bind-api-2.3.2.jar
+postgresql-42.3.3.jar
+checker-qual-3.5.0.jar
```
It's better to use one version if needed, you can pin the version in the root pom (dependency management)

zhangyuge1 replied:
The sub-dependencies of postgresql-42.3.3 include checker-qual-3.5.0.

CalvinKirs replied:
I know, but you can try it, we always hope for one version.

lcyyyyyy committed:
flink transform module support multi split,and add custome split function name (#2268)

EricJoy2048 committed:
[Docs]Add parquet support in doc (#2281)

TyrantLucifer committed:
[Connector-V2] Add file type check logic in hive connector (#2275)

hezean committed:
[CI][doc] Add build CI for document changes (#2280)

Hisoka-X committed:
[Doc][README] Fix contributor profile can't display (#2288)

songjianet committed:
[Feat][UI] Add the delete modal in the user manage. (#2293)

TyrantLucifer committed:
[Bug][Connector-V2] Fix avro conflicts (#2294)

* [Bug][Connector-V2] Fix avro conflicts (#2289)

* [Bug][Connector-V2] Fix avro conflicts (apache#2289)

* [Bug][Connector-V2] Fix dependency error (#2289)

EricJoy2048 committed:
add csv support in doc (#2295)

zhaomin1423 committed:
Avoid potential null pointer risk in HiveSinkWriter#snapshotState (#2302)

zhangyuge1 committed:
fix dependency issue

zhangyuge1 committed:
fix dependency issue

zhangyuge1 committed:
solve conflicts

ruanwenjun commented:
@zhangyuge1 Please rebase from the upstream and solve the conflicts.

zhangyuge1 committed:
Merge branch 'dev' of https://github.com/zhangyuge1/incubator-seatunnel into dev
